lib/chef/resource.rb

```diff
@@ -1012,6 +1010,75 @@ def default_action(action_name=NULL_ARG)
       def default_action=(action_name)
         default_action action_name
       end
+
+      #
+      # Define an action on this resource.
+      #
+      # The action is defined as a *recipe* block that will be compiled and then
+      # converged when the action is taken (when Resource is converged).  The recipe
+      # has access to the resource's attributes and methods, as well as the Chef
+      # recipe DSL.
+      #
+      # Resources in the action recipe may notify and subscribe to other resources
+      # within the action recipe, but cannot notify or subscribe to resources
+      # in the main Chef run.
+      #
+      # Resource actions are *inheritable*: if resource A defines `action :create`
+      # and B is a subclass of A, B gets all of A's actions.  Additionally,
+      # resource B can define `action :create` and call `super()` to invoke A's
+      # action code.
+      #
+      # @param name [Symbol] The action name to define.
```

Comment:
I think you should update this comment to include `The first non-:nothing action defined is the default action unless the default action is explicitly defined`.  You had some good wording for this somewhere...